indico/modules/events/persons/controllers.py

```diff
@@ -251,47 +249,70 @@ def _process(self):
+    """Preview an email with EventPersons associated placeholders."""
+
+    @use_kwargs({
+        'body': fields.String(required=True),
+        'subject': fields.String(load_default=None),
+    })
+    def _process(self, body, subject):
+        person = self.event.person_links[0]
+        email_body = replace_placeholders('event-persons-email', body, event=self.event, person=person)
+        email_subject = replace_placeholders('event-persons-email', subject, event=self.event, person=person)
+        tpl = get_template_module('events/persons/emails/custom_email.html', email_subject=email_subject,
+                                  email_body=email_body)
+        return {
+            'subject': tpl.get_subject(),
+            'body': tpl.get_body()
+        }
+
+
+class RHAPIEmailEventPersons(RHManageEventBase):
     """Send emails to selected EventPersons."""
 
     def _process_args(self):
-        self.no_account = request.args.get('no_account') == '1'
         RHManageEventBase._process_args(self)
+        self.no_account = request.args.get('no_account') == '1'
+        self.recipients = set(self._find_event_persons(request.args.getlist('person_id'),
+                                                       request.args.get('not_invited_only') == '1'))
+        self.recipients |= set(self._find_users(request.args.getlist('user_id')))
+        self.recipients |= set(self._find_role_members(request.args.getlist('role_id')))
 
-    def _process(self):
-        person_ids = request.form.getlist('person_id')
-        user_ids = request.form.getlist('user_id')
-        role_ids = request.form.getlist('role_id')
-        recipients = set(self._find_event_persons(person_ids, request.args.get('not_invited_only') == '1'))
-        recipients |= set(self._find_users(user_ids))
-        recipients |= set(self._find_role_members(role_ids))
+    def _process_GET(self):
         if self.no_account:
             tpl = get_template_module('events/persons/emails/invitation.html', event=self.event)
-            disabled_until_change = False
         else:
             tpl = get_template_module('events/persons/emails/generic.html', event=self.event)
-            disabled_until_change = True
-        form = EmailEventPersonsForm(person_id=person_ids, user_id=user_ids,
-                                     recipients=sorted(x.email for x in recipients), body=tpl.get_html_body(),
-                                     subject=tpl.get_subject(), register_link=self.no_account, event=self.event)
-        if form.validate_on_submit():
-            self._send_emails(form, recipients)
-            num = len(recipients)
-            flash(ngettext('Your email has been sent.', '{} emails have been sent.', num).format(num))
-            return jsonify_data()
-        return jsonify_form(form, disabled_until_change=disabled_until_change, submit=_('Send'), back=_('Cancel'))
-
-    def _send_emails(self, form, recipients):
+        # TODO: disabled until change
```

Comment:
The `disabled_until_change` functionality seems to be a bit complex to implement in the new form... Is it really necessary? Or since this is only to prevent distracted users sending the unaltered email template, maybe frontend-only validation would be enough?

Reply:
"disabled until change" is a purely client-side thing anyway (and it should come for free with the FinalModalForm if configured correctly)... it's just that previously it was usually enabled by passing this flag to the etmplate when rendering the form.

Reply:
how do I configure the FinalModalForm correctly for this? my idea was to add a validator to the body field and compare the value with the one from intialValues... is there an easier way to make a field invalid unless it's dirty?

Reply:
just pass the `disabledUntilChange` prop